modules/ROOT/pages/dataweave-cookbook-flat-file-read-and-write.adoc

```diff
@@ -0,0 +1,309 @@
+		Currency: payload.RQH.Currency
+	},
+	Batch: payload.Batch map ( batch , indexOfBatch ) -> {
+		BCH: {
+			"Record Type": batch.BCH."Record Type",
+			"Sequence Number": batch.BCH."Sequence Number",
+			"Batch Function": batch.BCH."Batch Function",
+			"Company Name": batch.BCH."Company Name",
+			"Unique Batch Identifier": batch.BCH."Unique Batch Identifier"},
+		TDR: batch.TDR map (tdr, indexOfTdr) ->
+			{
+			"Record Type": tdr.TDR."Record Type",
+			"Sequence Number": tdr.TDR."Sequence Number",
+			"Batch Function": tdr.TDR."Batch Function",
+			"Account Number": tdr.TDR."Account Number",
+			"Amount": tdr.TDR.Amount,
+			"Type": tdr.TDR.'Type'
+			},
+		BCF: {
+			"Record Type": batch.BCF."Record Type",
+			"Sequence Number": batch.BCF."Sequence Number",
+			"Batch Function": batch.BCF."Batch Function",
+			"Batch Transaction Amount": batch.BCF."Batch Transaction Amount",
+			"Type": batch.BCF.'Type',
+			"Batch Transaction Count": batch.BCF."Batch Transaction Count",
+			"Unique Batch Identifier": batch.BCF."Unique Batch Identifier"}
+		},
+	RQF: {
+		"Record Type": payload.RQF."Record Type",
+		"File Batch Count": payload.RQF."File Batch Count",
+		"File Transaction Count": payload.RQF."File Transaction Count",
+		"File Transaction Amount": payload.RQF."File Transaction Amount",
+		Type: payload.RQF.Type,
+		"Unique File Identifier": payload.RQF."Unique File Identifier"
+	}
+}
+]]>
+      </ee:set-payload>
+    	</ee:message>
+    </ee:transform>
+    <logger level="INFO" doc:name="Logger" message='#[payload]'/>
+    <file:write doc:name="Write flat file" config-ref="File_Config" path="${fileWritePath}" /> //<2>
+	</flow>
+</mule>
+----
+
+<1> Note that the Transform component specifies the following writer properties in the script header:
++
+* `output`: The output MIME type of the transformation, in this case, `application/flatfile`
+* `schemapath`: The writer property that specifies the path to the flat file schema (`.ffs` file)
+<2> Note that the File Write operation specifies the `path` property to define where to write the flat file
+
+After its execution, the example application generates the following output:
+
+----
+RQH201809011010A000000001USD
+BAT000001HACME RESEARCH                 A000000001
+BAT000000D          0000000000
+BAT000000D          0000000000
+BAT000004T0000120202CR000002A000000001
+BAT000005HAJAX EXPLOSIVES               A000000002
+BAT000000D          0000000000
+BAT000000D          0000000000
+BAT000008T0000003326DB000002A000000002
+RQF0002000008000000116876CRA000000001
+----
+
+== Steps to Test the Examples Locally
+
+If you want to test the examples on your end, follow these steps to create and configure your example Mule application:
+
+. Create a new *Mule Project* in Anypoint Studio.
+. Copy the XML code of the application you want to test into the *Configuration XML* view of your project.
+. xref:studio::add-modules-in-studio-to.adoc[Add the file connector module] to your project.
```

Comment:
```suggestion
. Use xref:studio::add-modules-in-studio-to.adoc[Add Modules] to add the Anypoint Connector for File (File Connector] to your project.
```

Reply:
Revised to refer to the connector by its official name while trying to make it clear that the Studio UI refers to it as a module...